Задание:
1.	Загрузить файл длиной не менее 2000 символов. 
2.	Составить программу, которая считает число уникальных слов в тексте (без критерия схожести)
3.	Составить программу, которая считает число гласных и согласных букв. 
4.	Составить программу, которая считает число предложений, их длину и число (количество) раз использования каждого слова в тексте (с критерием схожести, критерий схожести слов выбрать самостоятельно, например, spacy (en_core_web_sm) или расстояние Левенштейна). 
5.	Вывести 10 наиболее часто встречаемых слов. 

***
В данном блокноте выполняем пункт 3 задания.

***
Анализ по пункту 3 задания:
Предполагаем, что анализируется англоязычный текст.
Для подсчета числа гласных и согласных букв достаточно проходить текст символ за символом и проверять, обозначает ли символ гласный или согласный звук.
В случае положительного ответа увеличивать соответствующий счетчик на единицу. Поскольку число гласных мало, то можно задать только множество или строку гласных символов, а для согласных такое множество не создавать, а проверять, чтобы символ относился к алфавитным. Если символ входит в алфавит, но не относится к гласным, он будет обозначать согласную букву (хорошо, что в английском тексте нету букв типа мягкого или твердого знака, которые не обозначают ни гласные, ни согласные звуки).

Тогда план работы по пункту 3 следующий.

1. Загрузить текст из файла.
2. Перевести текст в нижний регистр.
3. Задать множество гласных букв.
4. Проходить текст символ за символом, проверяя его принадлежность к множеству гласных. В противном случае проверяем его принадлежность к алфавитным символам. В случае положительного ответа найденный символ будет давать согласную букву.
5. Вывести найденные значения числа согласных и гласных символов.

Загружаем текстовый файл:

In [2]:
# import

file_name = 'GospelJohn.txt' # file with text
try:
    f = open(file_name,"r") # open file for reading
    text = f.read()         # reading file 
    f.close()               # closing file
    print('Read successfully')
except:
    print('Error reading file!')

Read successfully


Переводим текст в нижний регистр.
Задаем строку гласных букв.
Проходим по всему тексту и проверяем, гласная или согласная буква.
Выводим результат

In [3]:
# to lower case:
text = text.lower()
vowels = 'aeiou' # vowel string
vowel_number = 0
consonant_number = 0
# find number of vowels and consonants
for c in text: 
    if c in vowels:
        vowel_number +=1
    elif c.isalpha():
        consonant_number +=1
    
print(f'Number of vowel letters: {vowel_number}')
print(f'Number of consonant letters: {consonant_number}')

Number of vowel letters: 1483
Number of consonant letters: 2509


Итак, нами был проделан анализ англоязычного текста в отношении количества гласных и согласных букв.

Выводы:
1. Целесообразно предварительно переводить текст в нижний регистр (или увеличивать набор гласных букв за счет прописных букв)
2. В англоязычном тексте размером 1000 слов (4102 знака без учета пробелов) гласных букв оказалось 1483 (около 36%), согласных букв - 2509 (около 61%). Согласных букв почти в два раза больше. Это соотношение может оказаться полезным в лингвистике, прикладной криптографии и других областях, где 
